activesupport/test/message_encryptor_test.rb

```diff
@@ -48,6 +48,16 @@ def test_signed_round_tripping
     assert_equal @data, @encryptor.decrypt_and_verify(message)
   end
 
+  def test_backwards_compat_for_64_bytes_key
+    # 64 bit key
+    secret = ["3942b1bf81e622559ed509e3ff274a780784fe9e75b065866bd270438c74da822219de3156473cc27df1fd590e4baf68c95eeb537b6e4d4c5a10f41635b5597e"].pack('H*')
+    # Encryptor with 32 bit key, 64 bit secret for verifier
+    encryptor = ActiveSupport::MessageEncryptor.new(secret[0..31], secret)
+    # Message generated with 64 bit key
+    message = "eHdGeExnZEwvMSt3U3dKaFl1WFo0TjVvYzA0eGpjbm5WSkt5MXlsNzhpZ0ZnbWhBWFlQZTRwaXE1bVJCS2oxMDZhYVp2dVN3V0lNZUlWQ3c2eVhQbnhnVjFmeVVubmhRKzF3WnZyWHVNMDg9LS1HSisyakJVSFlPb05ISzRMaXRzcFdBPT0=--831a1d54a3cda8a0658dc668a03dedcbce13b5ca"
+    assert_equal 'data', encryptor.decrypt_and_verify(message)[:some]
+  end
```

Comment:
Do we have a similar "known data" test up at the cookie jar / middleware level? I guess not, because it should've been failing between #25192 and #25602. It's probably worth adding one of those too... this level is important because we have an API contract with callers that we need to preserve, but arguably the most important of all is that we retain the ability to read an actual cookie value a previous version could have handed out.